modules/vmdatasource/src/xmpdatasource.cpp

```diff
@@ -83,11 +87,110 @@ void XMPDataSource::terminate()
 }
 
 XMPDataSource::XMPDataSource()
-  : IDataSource(), xmp(nullptr), metadataSource(nullptr)
+  : IDataSource(), xmp(nullptr), metadataSource(nullptr), compressor(nullptr)
 {
 
 }
 
+XMPDataSource::~XMPDataSource()
+{
+
+}
+
+void XMPDataSource::setCompressor(const vmf_string &id)
+{
+    if(!id.empty())
+    {
+        compressor = Compressor::create(id);
+        if(!compressor)
```

Comment:
if `Compressor::create(id)` throws an exception for unknown IDs, there is need to check the result.